pkg/doc/did/doc.go

```diff
@@ -712,6 +712,22 @@ func (doc *Doc) VerifyProof(suites ...verifier.SignatureSuite) error {
 	return v.Verify(docBytes)
 }
 
+// VerificationMethods returns all verification methods of DID Doc.
+func (doc *Doc) VerificationMethods() []VerificationMethod {
+	var verificationMethods []VerificationMethod
+
+	for i := range doc.PublicKey {
+		verificationMethods = append(verificationMethods, VerificationMethod{doc.PublicKey[i]})
+	}
+
+	for _, vm := range [...][]VerificationMethod{doc.Authentication, doc.CapabilityInvocation,
```

Comment:
References to other verification methods should be dereferenced, and I'm guessing the final list should contain unique entries (deduplicated)

Reply:
1. Added `Embedded` field to `VerificationMethod`. 
2. `DID.VerificationMethods()` returns a map with `VerificationRelationship` - it's a sort of "avoiding duplicates" approach.